File: stockAndWarehouseMasterFR/src/app/users/model/usuario.ts
import { Pedido } from "../../pedidos/model/pedido";
import { Producto } from "../../productos/model/producto";

export interface Usuario {
    id: number;
    nombre: string;
    username: string;
    contrasena: string;
    email: string;
    telefono: string;
    fechaAlta: Date;
    roles: Role[];
    photo?: Uint8Array;

    ubicacion: string;

    productosCesta: Producto[];
    listadoPedidos: Pedido[];
}

export interface Role {
    name : string;
}

File: stockAndWarehouseMaster/src/main/java/com/tfgAlberto/stockAndWarehouseMaster/usuarios/model/Usuario.java
package com.tfgAlberto.stockAndWarehouseMaster.usuarios.model;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
public class Usuario{

	private Long id;
	private String nombre;
	private String username;
	@JsonIgnore
	private String contrasena;
	private String email;
	private String telefono;
	private Date fechaAlta;
	private Set<RolUsuario> roles = new HashSet<RolUsuario>();
	private byte[] photo;

	private String ubicacion;
	
	public Usuario(String nombre, String username, String email, String contrasena) {
		this.nombre = nombre;
		this.username = username;
	    this.email = email;
	    this.contrasena = contrasena;
	}

}
